src/App.tsx

```diff
@@ -1,27 +1,102 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+enum TypeOfGoods {
+  None = '',
+  AllGoods = 'all-button',
+  FirstFiveGoods = 'first-five-button',
+  RedGoods = 'red-button',
+}
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [currentType, setCurrentType] = useState(TypeOfGoods.None);
+  const [isLoading, setLoading] = useState(false);
+  const [hasLoadingError, setLoadingError] = useState(false);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handleClickButton = (
+    getGoods: () => Promise<Good[]>,
+    typeOfGoods = TypeOfGoods.None,
+  ) => {
+    return () => {
+      if (currentType !== typeOfGoods) {
```

Comment:
```suggestion
      if (currentType === typeOfGoods && !hasLoadingError) {
        return;
      }
```

Simpler way to return if you don't need to do anything instead of wrapping code in block.